bundle/src/main/java/com/cognifide/cq/cqsm/core/scripts/ScriptFinderImpl.java

```diff
@@ -33,101 +35,91 @@
-	private List<Script> findAll(boolean skipIgnored, ResourceResolver resourceResolver) {
-		List<Script> scripts = new LinkedList<>();
-		for (String path : getSearchPaths()) {
-			Resource root = resourceResolver.getResource(path);
-			if (root != null) {
-				Iterator<Resource> children = root.listChildren();
-				scripts.addAll(getScripts(children, skipIgnored));
-			}
-		}
-		return scripts;
-	}
+  @Override
+  public List<Script> findAll(ResourceResolver resolver) {
+    return findAll(true, resolver);
+  }
 
-	@Override
-	public Script find(String path, ResourceResolver resolver) {
-		return find(path, true, resolver);
-	}
+  private List<Script> findAll(boolean skipIgnored, ResourceResolver resourceResolver) {
+    List<Script> scripts = new LinkedList<>();
+    for (String path : SEARCH_PATHS) {
+      Resource root = resourceResolver.getResource(path);
+      if (root != null) {
+        Iterator<Resource> children = root.listChildren();
+        scripts.addAll(getScripts(children, skipIgnored));
+      }
+    }
+    return scripts;
+  }
 
-	@Override
-	public Script find(String path, boolean skipIgnored, ResourceResolver resolver) {
-		Script result = null;
-		if (StringUtils.isNotEmpty(path) && (!skipIgnored || isNotIgnoredPath(path))) {
-			Resource resource = null;
-			if (path.contains(ROOT_PATH)) {
-				resource = resolver.getResource(path);
-			}
-			if (resource == null) {
-				path = path.startsWith("/") ? path.substring(1) : path;
-				for (String searchPath : getSearchPaths()) {
-					resource = resolver.getResource(searchPath + "/" + path);
-					if (resource != null) {
-						break;
-					}
-				}
-			}
-			if (resource != null) {
-				result = resource.adaptTo(ScriptImpl.class);
-			}
-		}
-		return result;
-	}
+  @Override
+  public Script find(String path, ResourceResolver resolver) {
+    return find(path, true, resolver);
+  }
 
-	private List<Script> getScripts(Iterator<Resource> scriptIterator, boolean skipIgnored) {
-		List<Script> scripts = new LinkedList<>();
-		while (scriptIterator.hasNext()) {
-			Resource resource = scriptIterator.next();
-			if (!skipIgnored || isNotIgnoredPath(resource.getPath())) {
-				Script script = resource.adaptTo(ScriptImpl.class);
-				if (script != null) {
-					scripts.add(script);
-				}
-			}
-		}
-		return scripts;
-	}
+  @Override
+  public Script find(String path, boolean skipIgnored, ResourceResolver resolver) {
+    Script result = null;
+    if (StringUtils.isNotEmpty(path) && (!skipIgnored || isNotIgnoredPath(path))) {
+      Resource resource = null;
+      if (path.contains(ROOT_PATH)) {
+        resource = resolver.getResource(path);
+      }
+      if (resource == null) {
+        path = path.startsWith("/") ? path.substring(1) : path;
+        for (String searchPath : SEARCH_PATHS) {
+          resource = resolver.getResource(searchPath + "/" + path);
+          if (resource != null) {
+            break;
+          }
+        }
+      }
+      if (resource != null) {

```

Comment:
shouldn't be just else for if statement above?

Reply:
No, it shouldn't. The resource could be left null in the above if statement. So current code handles more cases.